code/server/scripting/EmoteSystem/WebApi.cs

```diff
@@ -0,0 +1,28 @@
+﻿using CyberpunkSdk;
+using EmbedIO;
+using EmbedIO.Routing;
+
+namespace EmoteSystem;
+
+public record EmoteDto(string Username, string Emote);
+
+internal class EmoteController : PluginWebApiController
+{
+    private readonly Plugin _plugin;
+
+    internal EmoteController(Plugin plugin)
+    {
+        _plugin = plugin;
+    }
+
+    [Route(HttpVerbs.Get, "/")]
+    public EmoteDto GetLastEmote()
+    {
+        if (_plugin.LastEmote == null)
```

Comment:
This function is not thread safe, you read LastEmote, check the value, but it could change when returning, you need to cache the value.